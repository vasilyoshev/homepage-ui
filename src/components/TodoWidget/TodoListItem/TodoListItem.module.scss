.completed {
  text-decoration-line: line-through;
}

.todoText {
  cursor: pointer;
}
