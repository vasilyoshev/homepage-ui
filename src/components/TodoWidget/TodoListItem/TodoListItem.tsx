import React from 'react';
import { useDispatch } from 'react-redux';
import { TodoListItemProps } from 'interfaces';
import { removeTodo, completeTodo } from 'slices';
import { Checkbox, InputLabel, Typography, IconButton } from '@material-ui/core';
import DeleteIcon from '@material-ui/icons/Delete';
import styles from './TodoListItem.module.scss';

export const TodoListItem: React.FC<TodoListItemProps> = ({
  todo,
}) => {
  const dispatch = useDispatch();
  return (
    <InputLabel
      className={todo.completed ? styles.completed : ''}
    >
      <Checkbox
        checked={todo.completed}
        color="primary"
        edge="start"
        disableRipple
        onChange={() => dispatch(completeTodo(todo.id))}
      />
      <Typography
        className={styles.todoText}
        display="inline"
        color="primary">
        {todo.text}
      </Typography>
      <IconButton
        data-testid="removeTodoItem"
        aria-label="delete"
        onClick={() => dispatch(removeTodo(todo.id))}
      >
        <DeleteIcon
          fontSize="small"
        />
      </IconButton>
    </InputLabel>
  );
};
