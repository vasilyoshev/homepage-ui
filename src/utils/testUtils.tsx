import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

// TODO extract mocks to separate files, e.g. widgetsMock.js
const mockState = {
  loading: {},
  widgets: { widgetsProps: [] },
  user: {},
};
const createMockStore = configureStore([thunk]);
const store = createMockStore(mockState);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Providers = ({ children }: any) => (
  <Provider store={store}>
    <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
  </Provider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

export { customRender as render };
