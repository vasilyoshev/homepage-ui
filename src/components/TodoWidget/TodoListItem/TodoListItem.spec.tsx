import { TodoListItem } from './TodoListItem';
import { render, screen } from 'utils';
import userEvent from '@testing-library/user-event';

const mockData = [{
  text: 'Walk the dog',
  complete: false,
}];
describe('TodoListItem', () => {
  it('renders successfully', () => {
<<<<<<< HEAD
    const { asFragment } = render(<TodoListItem todo={mockData[0]} removeTodo={jest.fn()} toggleComplete={jest.fn()} />);
=======
    const { asFragment } = render(<TodoListItem todo={{} as Todo} />);
>>>>>>> main
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render todo item with checkbox and delete button', () => {
    render(<TodoListItem todo={mockData[0]} removeTodo={jest.fn()} toggleComplete={jest.fn()} />);
    const removeBtn = screen.getByTestId('removeTodoItem');
    const checkbox = screen.getByRole('checkbox', {name: /walk the dog/i});
    expect(checkbox).toBeInTheDocument();
    expect(removeBtn).toBeInTheDocument();
  });
  it('user can remove todo from list', () => {
    render(<TodoListItem todo={mockData[0]} removeTodo={jest.fn()} toggleComplete={jest.fn()} />);
    const removeBtn = screen.getByTestId('removeTodoItem');
    userEvent.type(removeBtn, 'Remove that todo');
    expect(screen.queryByText(/Remove that todo/i)).not.toBeInTheDocument();
  });
});
