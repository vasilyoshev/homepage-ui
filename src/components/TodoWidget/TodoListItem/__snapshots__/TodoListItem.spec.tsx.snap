// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem renders successfully 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-container css-oyakqy-MuiListItem-container"
  >
    <div
      class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding MuiListItem-secondaryAction listItem css-1ob1ktf-MuiListItem-root"
    >
      <div
        class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root PrivateSwitchBase-edgeStart css-zkqepn-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
      />
    </div>
    <div
      class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
    >
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</DocumentFragment>
`;
